### PR TITLE
[SPARK-40938][CONNECT][PYTHON][FOLLOW-UP] Fix SubqueryAlias without the child plan when constructing Connect proto in the Python client

### PR DESCRIPTION
**Pull request by amaliujia**
### What changes were proposed in this pull request?

The `SubqueryAlias` in `plan.py` does not set child plan when constructing Connect proto. This PR fixes that issue by setting up the input field during proto construction.

### Why are the changes needed?

Bug fix.

### Does this PR introduce _any_ user-facing change?

NO

### How was this patch tested?

UT

### CODE REVIEW

File: python/pyspark/sql/connect/plan.py [hunk 1]
```diff
@@ -712,6 +712,8 @@ def __init__(self, child: Optional["LogicalPlan"], alias: str) -> None:
 
     def plan(self, session: "RemoteSparkSession") -> proto.Relation:
         rel = proto.Relation()
+        if self._child is not None:
+            rel.subquery_alias.input.CopyFrom(self._child.plan(session))
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/plan.py:716]: Can we have an e2e test with `DataFrame.alias` since we're here?

Comment (amaliujia) [reply python/pyspark/sql/connect/plan.py:716]: Done.

BTW the e2e tests are slow. Not sure if we want to expand connect e2e test suites a lot eventually which could slow down the run job.

**Review by HyukjinKwon — APPROVED** [review]

File: python/pyspark/sql/connect/plan.py (hunk 1 above)
Comment (grundprinzip) [reply python/pyspark/sql/connect/plan.py:716]: The current e2e tests run about 16s on my laptop and the majority of the time is spend in starting spark. When we keep the restarts of Spark minimal, e2e tests should not be an issue when they make sense.

Comment (amaliujia) [reply python/pyspark/sql/connect/plan.py:716]: That will make sense then

**Comment by zhengruifeng** [comment]
Merged into master

**Pull request closed without merging**